code/datums/attack_style/weapon_styles/overhead.dm

```diff
@@ -0,0 +1,55 @@
+/datum/attack_style/melee_weapon/overhead
+
+/datum/attack_style/melee_weapon/overhead/get_swing_description(has_alt_style)
+	return "Comes down over the tile in the direction you are attacking. Always targets the head."
+
+/datum/attack_style/melee_weapon/overhead/finalize_attack(mob/living/attacker, mob/living/smacked, obj/item/weapon, right_clicking)
+	var/old_zone = attacker.zone_selected
+	attacker.zone_selected = BODY_ZONE_HEAD
+	. = ..()
+	attacker.zone_selected = old_zone
+
+/datum/attack_style/melee_weapon/overhead/attack_effect_animation(mob/living/attacker, obj/item/weapon, list/turf/affecting)
+	var/side_angle = prob(50) ? -15 : 15 // so it slightly goes either left -> right or right -> left
+	var/start_angle = -weapon.weapon_sprite_angle + get_angle(attacker, affecting[1]) + side_angle
+	var/image/attack_image = create_attack_image(attacker, weapon, affecting[1], start_angle)
+	var/matrix/final_matrix = turn(attack_image.transform, start_angle + (-2 * side_angle))
+	var/strike_dir = get_dir(attacker, affecting[1])
+	var/y_move = 0
+	var/x_move = 0
+	if(strike_dir & NORTH)
+		y_move += 3
+	else if(strike_dir & SOUTH)
+		y_move -= 3
+	if(strike_dir & EAST)
+		x_move += 3
+	else if(strike_dir & WEST)
+		x_move -= 3
+
+	attacker.do_attack_animation(affecting[1], no_effect = TRUE) // melbert todo
+	flick_overlay_global(attack_image, GLOB.clients, time_per_turf * 2.5)
```

Comment:
god I fucking hate this proc man it's implemented in the dumbest possible way
images are the worst

I don't think there's much to do about it here, but it worries me a bit

Reply:
I can try temp visuals

Reply:
unless it proves to be a massive issue it's likely ok for now. I wanna say vis_contents is better for most of these cases, but that requires objects, and it might be best to do that sort of thing all at once. idk